examples/ios/objc/Migration/Examples/Example_v3.h

```diff
@@ -0,0 +1,200 @@
+@end
+RLM_ARRAY_TYPE(Pet)
+
+@implementation Pet
++ (Pet* )petWithName:(NSString *)name kind:(enum Kind)kind {
+    Pet *pet = [[self alloc] init];
+    pet.name = name;
+    pet.kind = kind;
+    return pet;
+}
+- (enum Kind)kind {
+    return (Kind)_kindValue;
+}
+- (void)setKind:(enum Kind)kind {
+    _kindValue = kind;
+}
++ (NSArray *)requiredProperties {
+    return @[@"name", @"kindValue"];
+}
++ (NSArray *)ignoredProperties {
+    return @[@"kind"];
+}
+@end
+
+@interface Person : RLMObject
+@property NSString *fullName;
+@property int age;
+@property RLMArray<Pet *><Pet> *pets;
++ (Person* )personWithFullName:(NSString *)fullName age:(int)age;
+@end
+
+@implementation Person
++ (Person* )personWithFullName:(NSString *)fullName age:(int)age {
+    Person *person = [[self alloc] init];
+    person.fullName = fullName;
+    person.age = age;
+    return person;
+}
++ (NSArray *)requiredProperties {
+    return @[@"fullName", @"age"];
+}
+@end
+
+#pragma mark - Migration
+
+// Migration block to migrate from *any* previous version to this version.
+RLMMigrationBlock migrationBlock = ^(RLMMigration *migration, uint64_t oldSchemaVersion) {
+    if (oldSchemaVersion < 1) {
+        [migration enumerateObjects:Person.className block:^(RLMObject *oldObject, RLMObject *newObject) {
+            // combine name fields into a single field
+            newObject[@"fullName"] = [NSString stringWithFormat:@"%@ %@", oldObject[@"firstName"], oldObject[@"lastName"]];
+        }];
+    }
+    if (oldSchemaVersion < 2) {
+        [migration enumerateObjects:Person.className block:^(RLMObject *oldObject, RLMObject *newObject) {
+            // Add a pet to a specific person
+            if ([newObject[@"fullName"] isEqualToString:@"John Doe"]) {
+                // `Dog` was changed to `Pet` in v2 already, but we still need to account for this
+                // if upgrading from pre v2 to v3.
+                Pet *marley = [Pet petWithName:@"Marley" kind:dog];
+                Pet *lassie = [Pet petWithName:@"Lassie" kind:dog];
+                RLMArray<Pet *><Pet> *pets = newObject[@"pets"];
+                [pets addObject:marley];
+                [pets addObject:lassie];
+            } else if ([newObject[@"fullName"] isEqualToString:@"Jane Doe"]) {
+                Pet *toto = [Pet petWithName:@"Toto" kind:dog];
+                RLMArray<Pet *><Pet> *pets = newObject[@"pets"];
+                [pets addObject:toto];
+            }
+        }];
+        [migration createObject:Pet.className withValue:@[@"Slinkey", @1]];
+    }
+    if (oldSchemaVersion == 2) {
+        // This branch is only relevant for version 2. If we are migration from a previous
+        // version, we would not be able to access `dogs` since they did not exist back there.
+        // Migration from v0 and v1 to v3 is done in the previous blocks.
+        // Related issue: https://github.com/realm/realm-cocoa/issues/6263
```

Comment:
Is this issue still relevant?

Reply:
The issue itself is closed. This is the solution for it and I thought it might be nice to refer back to the original discussion to understand this part of the code better and why it is necessary.